linera-core/src/client.rs

```diff
@@ -1530,6 +1535,8 @@ where
         &self,
         operations: Vec<Operation>,
     ) -> Result<ExecuteBlockOutcome, ChainClientError> {
+        let block_mutex = Arc::clone(&self.state().preparing_block);
+        let _block_guard = block_mutex.lock_owned().await;
```

Comment:
Why not `let block_mutex = self.state().preparing_block.clone();` ?

Reply:
A good question!  I read it off the [`tokio::sync::Mutex` documentation](https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html#method.blocking_lock_owned) but if I were to take a guess, it's to make sure that a `clone()` doesn't sneakily change its meaning if the type of the variable were to change.  In general, it's advised that functions on types that implement `Deref` not take `self`, so as to avoid any potential ambiguity with the target type.

Reply:
Well I was not far off, [here it is in the book](https://doc.rust-lang.org/book/ch15-04-rc.html#using-rct-to-share-data):

> We could have called `a.clone()` rather than `Rc::clone(&a)`, but Rust’s convention is to use `Rc::clone` in this case. The implementation of `Rc::clone` doesn’t make a deep copy of all the data like most types’ implementations of `clone` do. The call to `Rc::clone` only increments the reference count, which doesn’t take much time. Deep copies of data can take a lot of time. By using `Rc::clone` for reference counting, we can visually distinguish between the deep-copy kinds of clones and the kinds of clones that increase the reference count. When looking for performance problems in the code, we only need to consider the deep-copy clones and can disregard calls to `Rc::clone`.

I think I prefer my rationale, but either way it is a standard convention.